### PR TITLE
bpo-41410: Fix outdated info in mkstemp docs

### PR DESCRIPTION
**Pull request by xrisk**
https://bugs.python.org/issue41410

Automerge-Triggered-By: @ericvsmith

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@xrisk

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by rhettinger — APPROVED** [review]

**Review by ericvsmith — CHANGES_REQUESTED** [review]
One small point about truthiness.

**Comment by xrisk** [comment]
I have made the requested changes; please review again

@ericvsmith I didn’t quite get what you. Is this what you meant?

File: Doc/library/tempfile.rst [hunk 1]
```diff
@@ -175,9 +175,9 @@ The module defines the following user-callable items:
    If you want to force a bytes return value with otherwise default behavior,
    pass ``suffix=b''``.
 
-   If *text* is specified, it indicates whether to open the file in binary
-   mode (the default) or text mode.  On some platforms, this makes no
-   difference.
+   If *text* is specified and ``True`` (or a truthy value), the file is
```
Comment (ericvsmith) [inline Doc/library/tempfile.rst]: I'd skip the mention of `True`, and just say "If *text* is specified and true, the file is" ...

I think the way it is is too much jargon for beginners. I quick glance through some of the other docs doesn't show any mention of `True` or truthiness, just "true".

**Review by ericvsmith — CHANGES_REQUESTED** [review]
When this small issue is addressed and the CLA is signed, we can merge this.

File: Doc/library/tempfile.rst (hunk 1 above)
Comment (rhettinger) [reply Doc/library/tempfile.rst]: I concur.

**Comment by xrisk** [comment]
I have made the requested changes; please review again.

**Review by ericvsmith — COMMENTED** [review]
This looks good, thanks. Now it's just waiting on the CLA (Contributor License Agreement).

**Comment by xrisk** [comment]
Hi do I need to do anything else here?

**Review by ericvsmith — APPROVED** [review]

**Pull request merged**